cartoframes/carto_vl/layer/layer.py

```diff
@@ -22,24 +31,133 @@ class Layer(QueryLayer):
                 api_key='default_public'
             )
 
-            vl.Map(
-                [vl.Layer('populated_places')],
+            vl.Map([
+                vl.Layer(
+                    source=vl.source.SQL('SELECT * FROM populated_places WHERE adm0name = \'Spain\''),
+                    style={
+                        'color': 'red'
+                    }
+                )],
                 context=context
             )
     """
+
     def __init__(self,
-                 query,
+                 source,
                  style=None,
                  variables=None,
                  interactivity=None,
                  legend=None):
 
-        self.table_source = query
+        self.orig_query = source.query
+        self.is_basemap = False
+        self.source = source  # TO DO check instance of Source
+        self.bounds = source.bounds
+        self.style = _parse_style_properties(style)
+        self.variables = _parse_variables(variables)
+        self.interactivity = _parse_interactivity(interactivity)
+        self.legend = legend
+        self.viz = _set_viz(self.variables, self.style)
+
+
+def _convstr(obj):
+    """convert all types to strings or None"""
+    return str(obj) if obj is not None else None
+
+
+def _to_camel_case(text):
+    """convert properties to camelCase"""
+    components = text.split('-')
+    return components[0] + ''.join(x.title() for x in components[1:])
 
-        super(Layer, self).__init__(
-            'SELECT * FROM {}'.format(query),
-            style=style,
-            variables=variables,
-            interactivity=interactivity,
-            legend=legend
+
+def _parse_style_properties(style):
+    """Adds style properties to the styling"""
+    if style is None:
+        return ''
+    elif isinstance(style, dict):
+        return _parse_style_properties_dict(style)
+    elif isinstance(style, (tuple, list)):
+        return _parse_style_properties_list(style)
+    elif isinstance(style, str):
+        return style
+    else:
+        raise ValueError('`style` must be a dictionary, a list or a string')
+
+
+def _parse_style_properties_dict(style):
+    return '\n'.join(
```

Comment:
Should we throw warnings if a user supplies a style property that is not in the `STYLE_PROPERTIES` set?

Reply:
CARTO VL already shows a warning in the console: `console.warn(`Property '${key}' is not supported`)`, I don't know if this is enough or if we should show throw an error from CARTOFrames.